dot-net-sdk/EppoClient.cs

```diff
@@ -30,35 +31,46 @@ public class EppoClient
         _fetchExperimentsTask = fetchExperimentsTask;
     }
 
-    public JObject? GetJsonAssignment(string subjectKey, string flagKey, SubjectAttributes? subjectAttributes = null)
+    private HasEppoValue _typeCheckedAssignment(string flagKey, string subjectKey, Subject subjectAttributes, EppoValueType expectedValueType, object defaultValue) {
+        var result = GetAssignment(flagKey, subjectKey, subjectAttributes);
+        var eppoDefaultValue = new HasEppoValue(defaultValue);
+        if (HasEppoValue.IsNullValue(result)) return eppoDefaultValue;
+        var assignment = result!;
+        if (assignment.Type != expectedValueType) {
+            Logger.Warn($"[Eppo SDK] Expected type {expectedValueType} does not match parsed type {assignment.Type}");
+            return eppoDefaultValue;
+        }
+        return assignment;
+    }
+    public JObject GetJsonAssignment(string flagKey, string subjectKey, Subject subjectAttributes, JObject defaultValue)
     {
-        return GetAssignment(subjectKey, flagKey, subjectAttributes ?? new SubjectAttributes())?.JsonValue();
+        return _typeCheckedAssignment(flagKey, subjectKey, subjectAttributes, EppoValueType.JSON, defaultValue).JsonValue();
     }
 
-    public bool? GetBoolAssignment(string subjectKey, string flagKey, SubjectAttributes? subjectAttributes = null)
+    public bool GetBoolAssignment(string flagKey, string subjectKey, Subject subjectAttributes, bool defaultValue)
```

Comment:
We're doing a push to move to consistent naming across all SDKs and use `Boolean` instead of `Bool`
```suggestion
    public bool GetBooleanAssignment(string flagKey, string subjectKey, Subject subjectAttributes, bool defaultValue)
```